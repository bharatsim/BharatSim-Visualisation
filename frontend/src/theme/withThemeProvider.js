import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
