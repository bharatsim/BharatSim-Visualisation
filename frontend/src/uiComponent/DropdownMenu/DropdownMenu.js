import Box from '@material-ui/core/Box';
import React from 'react';
import MenuItem from '@material-ui/core/MenuItem';
import { Typography } from '@material-ui/core';
import Menu from '@material-ui/core/Menu';
import PropTypes from 'prop-types';
import Divider from "@material-ui/core/Divider";
import useMenuStyles from './dropdownMenuCss';

function DropdownMenu({ anchorEl, closeMenu, menuItems }) {
  const menuClasses = useMenuStyles();
  return (
    <Menu
      id="simple-menu"
      anchorEl={anchorEl}
      keepMounted
      open={Boolean(anchorEl)}
      onClose={closeMenu}
      classes={{ paper: menuClasses.menuPaper }}
      PopoverClasses={{
        paper: menuClasses.popoverPaper,
      }}
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'right',
      }}
      transformOrigin={{
        vertical: 'top',
        horizontal: 'right',
      }}
      getContentAnchorEl={null}
    >
      {menuItems.map(({ icon, label, onClick, dataTestId, withDivider = false }) => {
        return (
          <>
            { withDivider && <Divider />}
            <MenuItem
              onClick={onClick}
              classes={{ root: menuClasses.root }}
              data-testid={dataTestId}
              key={dataTestId}
            >
              <Box className={menuClasses.menuOption}>
                {icon}
                <Box ml={4}>
                  <Typography variant="body2">{label}</Typography>
                </Box>
              </Box>
            </MenuItem>
          </>
        );
      })}
    </Menu>
  );
}

DropdownMenu.defaultProps = {
  anchorEl: null,
};

DropdownMenu.propTypes = {
  anchorEl: PropTypes.oneOfType([PropTypes.shape({})]),
  closeMenu: PropTypes.func.isRequired,
  menuItems: PropTypes.arrayOf(
    PropTypes.shape({
      icon: PropTypes.element.isRequired,
      label: PropTypes.string.isRequired,
      onClick: PropTypes.func.isRequired,
      dataTestId: PropTypes.string.isRequired,
      withDivider: PropTypes.bool,
    }),
  ).isRequired,
};

export default DropdownMenu;
