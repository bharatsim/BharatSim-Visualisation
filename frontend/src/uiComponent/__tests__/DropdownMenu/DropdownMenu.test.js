import React from 'react';
import { render } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import DropdownMenu from '../../dropdownMenu/DropdownMenu';
import withThemeProvider from '../../../theme/withThemeProvider';
import deleteIcon from '../../../assets/images/delete.svg';

describe('<DropdownMenu />', () => {
  const DropdownMenuComponent = withThemeProvider(DropdownMenu);
  it('should match snapshot', () => {
    const { container } = render(
      <DropdownMenuComponent
        anchorEl={null}
        closeMenu={jest.fn()}
        menuItems={[
          {
            label: 'Delete Dashboard',
            icon: <img src={deleteIcon} alt="delete-logo" />,
            onClick: jest.fn(),
            dataTestId: `delete-option-test-id`,
          },
        ]}
      />,
    );
    expect(container).toMatchSnapshot();
  });
  it('should call onclick action of menu item on click of option', () => {
    const mockClickFunction = jest.fn();
    const { getByTestId } = render(
      <DropdownMenuComponent
        anchorEl={document.createElement('div')}
        closeMenu={jest.fn()}
        menuItems={[
          {
            label: 'Delete Dashboard',
            icon: <img src={deleteIcon} alt="delete-logo" />,
            onClick: mockClickFunction,
            dataTestId: `delete-option-test-id`,
          },
        ]}
      />,
    );
    const option = getByTestId('delete-option-test-id');
    fireEvent.click(option);
    expect(mockClickFunction).toHaveBeenCalled();
  });
});
