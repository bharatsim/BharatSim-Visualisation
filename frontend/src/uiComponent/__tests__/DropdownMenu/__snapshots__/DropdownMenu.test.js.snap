// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownMenu /> should match snapshot 1`] = `<div />`;
